packages/bot/src/scratch/index.js

```diff
@@ -209,13 +209,17 @@ class DBot {
         this.valueInputLimitationsListener({}, true);
 
         const all_blocks  = this.workspace.getAllBlocks(true);
-        const error_block = all_blocks.find(block => block.is_error_highlighted && !block.disabled);
-
-        if (error_block) {
-            const { run_panel } = ScratchStore.instance.root_store;
-            const message       = localize('The block(s) highlighted in red are missing input values. Please update them and click "Run bot".');
-            this.workspace.centerOnBlock(error_block.id);
-            run_panel.showErrorMessage(message);
+        const error_blocks = all_blocks
+            .filter(block => block.is_error_highlighted && !block.disabled)
+            .filter((block, index, self) => index === self.findIndex(b => b.error_message === block.error_message));
```

Comment:
Could you describe the logic here, I'm not sure what's happening here.

Reply:
I'm trying to filter out duplicate error message so that it won't show the same message over in journal.

Reply:
Could you add a comment to the code here?